packages/engine/src/interaction/ui/InteractiveModalView.tsx

```diff
@@ -0,0 +1,433 @@
+import React from 'react'
+import { createState } from '@hookstate/core'
+import { createXRUI } from '../../xrui/functions/createXRUI'
+import { useXRUIState } from '@xrengine/engine/src/xrui/functions/useXRUIState'
+import { NavigateNext, NavigateBefore } from '@mui/icons-material'
+
+let callback: any
+
+export function createInteractiveModalView(data: any) {
+  return createXRUI(InteractiveModalView, createInteractiveModalState(data))
+}
+
+export function connectCallback(cb: any) {
+  callback = cb
+}
+
+function createInteractiveModalState(data: any) {
+  const totalMediaUrls: any[] = []
+  for (let url of data.interactionImages) {
+    totalMediaUrls.push({
+      type: 'image',
+      path: url
+    })
+  }
+  for (let url of data.interactionVideos) {
+    totalMediaUrls.push({
+      type: 'video',
+      path: url
+    })
+  }
+  for (let url of data.interactionModels) {
+    totalMediaUrls.push({
+      type: 'model',
+      path: url
+    })
+  }
+  let entityIndex = 0
+  if (data.interactionUserData && data.interactionUserData.entity) {
+    entityIndex = data.interactionUserData.entity
+  }
+  return createState({
+    title: data.interactionName,
+    description: data.interactionDescription,
+    images: data.interactionImages,
+    videos: data.interactionVideos,
+    urls: data.interactionUrls,
+    models: data.interactionModels,
+    themeIndex: data.interactionThemeIndex,
+    mediaIndex: data.mediaIndex ? data.mediaIndex : 0,
+    entityIndex,
+    totalMediaUrls
+  })
+}
+
+type InteractiveDetailState = ReturnType<typeof createInteractiveModalState>
+
+const InteractiveModalView = () => {
+  const detailState = useXRUIState() as InteractiveDetailState
+
+  const nextClick = () => {
+    let value = detailState.mediaIndex.value
+    value++
+    if (value > detailState.totalMediaUrls.value.length) value = 0
+    detailState.mediaIndex.set(value)
+    if (callback) {
+      callback({
+        mediaIndex: value,
+        mediaData: detailState.totalMediaUrls.value,
+        entityIndex: detailState.entityIndex.value
+      })
+    }
+  }
+
+  const beforeClick = () => {
+    let value = detailState.mediaIndex.value
+    value--
+    if (value < 0) value = detailState.totalMediaUrls.value.length - 1
+    detailState.mediaIndex.set(value)
+    if (callback) {
+      callback({
+        mediaIndex: value,
+        mediaData: detailState.totalMediaUrls.value,
+        entityIndex: detailState.entityIndex.value
+      })
+    }
+  }
+
+  const linkClick = (urls) => {
+    for (let url of urls) {
+      window.open(url, '_blank')!.focus()
+    }
+  }
+
+  const renderMedia = (width, height, position?) => {
```

Comment:
FYI, if you want to add :hover styles, you can use `<style dangerouslySetInnerHTML={}></style>` to add normal CSS. Hover styles will be rendered automatically.